hbase/src/main/scala/akka/stream/alpakka/hbase/internal/HBaseFlowStage.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.hbase.internal
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.stream.alpakka.hbase.HTableSettings
+import akka.stream.stage._
+import akka.stream.{ Attributes, FlowShape, Inlet, Outlet }
+import org.apache.hadoop.hbase.client.Table
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ Future, Promise }
+import scala.util.control.NonFatal
+import scala.util.{ Failure, Success, Try }
+
+private[hbase] class HBaseFlowStage[A](settings: HTableSettings[A])
+    extends GraphStage[FlowShape[A, Future[A]]]
+    with HBaseCapabilities {
+
+  private val in = Inlet[A]("messages")
+  private val out = Outlet[Future[A]]("result")
+
+  override val shape = FlowShape(in, out)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = {
+
+    implicit val connection = connect(settings.conf)
+
+    val table: Table = getOrCreateTable(settings.tableName, settings.columnFamilies)
```

Comment:
both these two does blocking network logic, now it happens during materialization stopping the graph from complete materialization until the blocking call is completed, in general we rather try to isolate such logic inside of the graph stage logic and let the materialization complete and have the logic backpressure until a connection has been made. The blocking logic needs to be isolated on a separate dispatcher for this or it will starve the default dispatcher. (I think I mentioned this in a previous review)